ports/psoc6/machine_i2s.c

```diff
@@ -172,8 +148,6 @@ uint8_t get_word_byte_size(machine_i2s_obj_t *self) {
         return 1;
```

Comment:
The resolution can be only 16bits or 32 bits right? In that case, could we simplify the function further?

Reply:
Technically you are right. This is there because the cyhal_i2s allows any bits resolution (word length), and then they will round the frames to 8, 16 or 32 bits (channel length). But then the words will be fitted into 16 or 32 frames in the micropython API. 
So those 8 bits would make sense for DMA transfer purposes of the cyhal_i2s API. But honestly I haven´t tested other resolutions. I have kept the option of passing any bits resolution also the constructor. 

The main reason I left the option of passing any bits resolution open is because there are some i2s devices not using the typical16 or 32 (i.e. the infineon IM69D microphone  has 20 bits resolution). But I used 32 bits for recording a .wav and it was working as well, just padding with 0s the 12 MSbits.

Therefore, (A) we can simplify it by only allowing 16 and 32 and word length = channel length, and remove any other bit resolution option. (B) Or leave as it is, and do some more testing in the future for 8 bits, or intermediate bit resolution. What do you think?